src/StackExchange.Redis/AzureMaintenanceEvent.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using static StackExchange.Redis.ConnectionMultiplexer;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// Azure node maintenance event details
+    /// </summary>
+    public class AzureMaintenanceEvent
+    {
+        internal AzureMaintenanceEvent(string message)
+        {
+            RawMessage = message;
+            try
+            {
+                var info = message?.Split('|');
+                for (int i = 0; i < info?.Length / 2; i++)
+                {
+                    string key = null, value = null;
+                    if (2 * i < info.Length) { key = info[2 * i].Trim(); }
+                    if (2 * i + 1 < info.Length) { value = info[2 * i + 1].Trim(); }
+                    if (!string.IsNullOrEmpty(key) && !string.IsNullOrEmpty(value))
+                    {
+                        switch (key.ToLowerInvariant())
+                        {
+                            case "notificationtype":
+                                NotificationType = value;
+                                break;
+
+                            case "starttimeinutc":
+                                if (DateTime.TryParse(value, out DateTime startTime))
+                                {
+                                    StartTimeUtc = DateTime.SpecifyKind(startTime, DateTimeKind.Utc);
+                                }
+                                break;
+
+                            case "isreplica":
+                                bool.TryParse(value, out IsReplica);
+                                break;
+
+                            case "ipaddress":
+                                IPAddress.TryParse(value, out IpAddress);
+                                break;
+
+                            case "sslport":
+                                Int32.TryParse(value, out var port);
+                                SSLPort = port;
+                                break;
+
+                            case "nonsslport":
+                                Int32.TryParse(value, out var nonsslport);
+                                NonSSLPort = nonsslport;
+                                break;
+
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+            catch { }
+        }
+
+        internal async static Task AddListenerAsync(ConnectionMultiplexer multiplexer, LogProxy logProxy)
+        {
+            try
+            {
+                var sub = multiplexer.GetSubscriber();
+                if (sub == null)
+                {
+                    logProxy?.WriteLine("Failed to GetSubscriber for AzureRedisEvents");
+                    return;
+                }
+
+                await sub.SubscribeAsync("AzureRedisEvents", (channel, message) =>
```

Comment:
Minor suggestion: recommend we make this channel name (AzureRedisEvents) a private constant up top, so that the primary entry point for this whole deal isn't down in and a bit easier for others to spot later.